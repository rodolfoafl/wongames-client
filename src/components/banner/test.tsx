import { render, screen } from '@testing-library/react'

import Banner from '.'
import Button from '@/components/button'

const props = {
  image:
    'https://images.unsplash.com/photo-1741893041975-94a0e8656209?q=80&w=1742&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D',
  title: 'Banner Title',
  subtitle:
    'Lorem ipsum dolor, sit amet consectetur adipisicing elit. Nihil, cum!'
}

describe('<Banner />', () => {
  it('should render the banner', () => {
    const { container } = render(<Banner {...props} />)

    const image = screen.getByRole('img')
    expect(image).toBeInTheDocument()
    expect(image).toHaveAttribute('src')

    const title = screen.getByText(props.title)
    expect(title).toBeInTheDocument()

    const subtitle = screen.getByText(props.subtitle)
    expect(subtitle).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the children if passed', () => {
    render(
      <Banner {...props}>
        <Button>Buy now</Button>
      </Banner>
    )

    const button = screen.getByRole('button', { name: /buy now/i })
    expect(button).toBeInTheDocument()
  })
})
