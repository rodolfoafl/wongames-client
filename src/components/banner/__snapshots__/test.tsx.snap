// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> should render the banner 1`] = `
<div
  class="relative md:shadow-md"
>
  <img
    alt="Banner Title"
    aria-label="Banner Title"
    class="bg-light-gray h-[14.375rem] w-full object-cover object-center md:h-[36.25rem]"
    data-nimg="1"
    decoding="async"
    height="480"
    loading="lazy"
    role="img"
    src="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1741893041975-94a0e8656209%3Fq%3D80%26w%3D1742%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=1200&q=75"
    srcset="/_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1741893041975-94a0e8656209%3Fq%3D80%26w%3D1742%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fimages.unsplash.com%2Fphoto-1741893041975-94a0e8656209%3Fq%3D80%26w%3D1742%26auto%3Dformat%26fit%3Dcrop%26ixlib%3Drb-4.0.3%26ixid%3DM3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%253D%253D&w=1200&q=75 2x"
    style="color: transparent;"
    width="600"
  />
  <div
    class="flex w-full flex-col gap-y-2 bg-[#000000b3] p-6 md:absolute md:bottom-0 md:left-0 md:rounded-br-sm md:rounded-bl-sm md:p-10"
  >
    <h2
      class="text-lg leading-none font-bold text-white md:text-[1.75rem]"
    >
      Banner Title
    </h2>
    <p
      class="[&_strong]:text-primary text-sm leading-none font-normal text-white md:text-lg"
    >
      Lorem ipsum dolor, sit amet consectetur adipisicing elit. Nihil, cum!
    </p>
  </div>
</div>
`;
